plugins/scroll-options/src/ScrollMetricsManager.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * The MetricsManager reports various metrics about the workspace. This version
+ * of the MetricsManager adds optional cacheing of the workspace content
+ * metrics. Cached content metrics are useful if, for example, you are in the
+ * middle of dragging a block, but want to get the bounds of the content area as
+ * if you hadn't yet picked up the block.
+ *
+ * To use the cached value of the content metrics instead of calculating new
+ * ones, set `stopCalculating` to true before calling `getMetrics` and false
+ * when the metrics can be recalculated again.
+ */
+export class ScrollMetricsManager extends Blockly.MetricsManager {
+  /** @override */
+  constructor(workspace) {
+    super(workspace);
+
+    /**
+     * Whether to stop recalculating content metrics and used the cached value
+     * instead. Note that if there are no cached metrics, they will be
+     * recalculated even if `stopCalculating` is true.
+     * @type {boolean}
+     */
+    this.stopCalculating = false;
+  }
+
+  /** @override */
+  getMetrics() {
```

Comment:
Unfortunately jus thought of this :/ but what if instead of only stopping updating for the content metrics we stopped updating all the metrics? It would be something like: 

```
if we are calculating
    this.metrics = super.getMetrics();
return this.metrics
```
My motivation behind this is that anytime we call super we decrease the amount of maintenance the function requires.

Reply:
As discussed offline, we can't cache all the metrics as some of them still need to update (such as view and scroll). However, changed to only override getContentMetrics method, which is much simpler.

Also, changed the name of the property that controls this behavior to be more clear, ptal.